spark-knn-core/src/main/scala/org/apache/spark/ml/classification/KNNClassifier.scala

```diff
@@ -76,7 +76,7 @@ with KNNParams with HasWeightCol with Logging {
   /** @group setParam */
   def setSeed(value: Long): this.type = set(seed, value)
 
-  override protected def train(dataset: DataFrame): KNNClassificationModel = {
+  override protected def train(dataset: Dataset[_]): KNNClassificationModel = {
```

Comment:
In built-in tools, the parameter of this method is always a DataFrame, same for fit and transform